clients/src/main/java/org/apache/kafka/common/utils/Exit.java

```diff
@@ -119,4 +119,51 @@ public static void resetHaltProcedure() {
     public static void resetShutdownHookAdder() {
         shutdownHookAdder = DEFAULT_SHUTDOWN_HOOK_ADDER;
     }
+
+    public void exitOrThrow(int statusCode) {
+        exit(statusCode, null);
+    }
+
+    public abstract void exitOrThrow(int statusCode, String message);
+
+    public void haltOrThrow(int statusCode) {
+        halt(statusCode, null);
+    }
+
+    public abstract void haltOrThrow(int statusCode, String message);
+
+    public abstract void addShutdownRunnable(String name, Runnable runnable);
```

Comment:
Could we have brief Javadocs on these? Beyond the obvious, it'd be nice to note when callers could expect the `OrThrow` behavior to take place.